microsoft/testsuites/performance/networkperf.py

```diff
@@ -196,15 +247,137 @@ def perf_tcp_pps(
+                    buffer_size = int(1048576 / 1024)
+                else:
+                    buffer_size = int(65536 / 1024)
+                if udp_mode:
+                    buffer_size = int(1024 / 1024)
+                server_result = server_ntttcp.run_as_server_async(
+                    server_nic_name,
+                    ports_count=num_threads_p,
+                    buffer_size=buffer_size,
+                    dev_differentiator=dev_differentiator,
+                    udp_mode=udp_mode,
+                )
+                client_lagscope_process = client_lagscope.run_as_client_async(
+                    server_ip=server.internal_address,
+                    ping_count=0,
+                    run_time_seconds=120,
+                    print_histogram=False,
+                    print_percentile=False,
+                    histogram_1st_interval_start_value=0,
+                    length_of_histogram_intervals=0,
+                    count_of_histogram_intervals=0,
+                    dump_csv=False,
+                )
+                client_ntttcp_result = client_ntttcp.run_as_client(
+                    client_nic_name,
+                    server.internal_address,
+                    buffer_size=buffer_size,
+                    threads_count=num_threads_n,
+                    ports_count=num_threads_p,
+                    dev_differentiator=dev_differentiator,
+                    udp_mode=udp_mode,
+                )
+                server_ntttcp_result = server_result.wait_result()
+                server_result_temp = server_ntttcp.create_ntttcp_result(
+                    server_ntttcp_result
+                )
+                client_result_temp = client_ntttcp.create_ntttcp_result(
+                    client_ntttcp_result, role="client"
+                )
+                client_sar_result = client_lagscope_process.wait_result()
+                client_average_latency = client_lagscope.get_average(client_sar_result)
+                if udp_mode:
+                    perf_ntttcp_message_list.append(
+                        client_ntttcp.create_ntttcp_udp_performance_message(
+                            server_result_temp,
+                            client_result_temp,
+                            str(test_thread),
+                            buffer_size,
+                        )
+                    )
+                else:
+                    perf_ntttcp_message_list.append(
+                        client_ntttcp.create_ntttcp_tcp_performance_message(
+                            server_result_temp,
+                            client_result_temp,
+                            client_average_latency,
+                            str(test_thread),
+                            buffer_size,
+                        )
+                    )
+            information: Dict[str, str] = environment.get_information()
+            for ntttcp_message in perf_ntttcp_message_list:
```

Comment:
As mentioned in previous comments, the messages can be sent when it's generated. It doesn't need to be sent at the end of test cases together.